wolfcrypt/src/asn.c

```diff
@@ -10562,20 +10562,25 @@ int PemToDer(const unsigned char* buff, long longSz, int type,
             }
             /* decrypt the key */
             else {
-                ret = wc_BufferKeyDecrypt(info, der->buffer, der->length,
-                    (byte*)password, passwordSz, WC_MD5);
+                if (passwordSz == 0) {
+                    /* The key is encrypted but does not have a password */
+                    ret = ASN_PARSE_E;
```

Comment:
This is better than no explicit error for a missing password but not really accurate, and suggests the key file/buffer is corrupt.  Please pick a more appropriate error or create a new one.  And add a WOLFSSL_MSG() to help with debugging.

Reply:
Fixed. I used the appropriately named `NO_PASSWORD` error.